core/src/services/redis/backend.rs

```diff
@@ -326,8 +342,43 @@ impl Adapter {
     }
 }
 
+#[self_referencing]
+struct RedisAsyncConnIter<'a> {
+    conn: bb8::PooledConnection<'a, RedisConnectionManager>,
+
+    #[borrows(mut conn)]
+    #[not_covariant]
+    iter: AsyncIter<'this, String>,
+}
+
+#[self_referencing]
+pub struct RedisScanner {
+    pool: bb8::Pool<RedisConnectionManager>,
```

Comment:
We could extract `RedisCore` as other services do and use `Arc<RedisCore>` here to avoid duplicate calls like `conn_from_pool`.